subxt/src/storage/storage_key.rs

```diff
@@ -50,18 +55,24 @@ impl StorageKey for () {
     }
 
     fn decode_from_bytes(
-        _cursor: &mut &[u8],
+        bytes: &mut &[u8],
         hashers_and_ty_ids: &mut &[(StorageHasher, u32)],
-        _metadata: &Metadata,
+        metadata: &Metadata,
     ) -> Result<Self, Error> {
-        if hashers_and_ty_ids.is_empty() {
-            return Err(StorageAddressError::WrongNumberOfHashers {
-                hashers: 0,
-                fields: 1,
+        // If no hashers, we just do nothing (erroring if ).
```

Comment:
When there _is_ a hasher, this also needs to skip over any key bytes if they exist (which is what `consume_hash_returning_key_bytes` does.

When there is not a hasher, we just exit happily unless there are leftover bytes, which would be unexpected.